tests/test_mathematical.py

```diff
@@ -424,6 +424,39 @@ def test_out_dtype(self, arr_dt, out_dt, dtype):
         assert_array_equal(expected, result)
         assert result is iout
 
+    @testing.with_requires("numpy>=2.0.0")
```

Comment:
based on NumPy release notes:
```suggestion
    @testing.with_requires("numpy>=2.1.0")
```